src/components/sbb-notification/sbb-notification.tsx

```diff
@@ -0,0 +1,260 @@
+import {
+  Component,
+  ComponentInterface,
+  Element,
+  Event,
+  EventEmitter,
+  h,
+  Host,
+  JSX,
+  Method,
+  Prop,
+  State,
+  Watch,
+} from '@stencil/core';
+import { InterfaceTitleAttributes } from '../sbb-title/sbb-title.custom';
+import { i18nCloseNotification } from '../../global/i18n';
+import {
+  createNamedSlotState,
+  documentLanguage,
+  HandlerRepository,
+  languageChangeHandlerAspect,
+  namedSlotChangeHandlerAspect,
+} from '../../global/helpers';
+import { InterfaceNotificationAttributes } from './sbb-notification.custom';
+import { AgnosticResizeObserver as ResizeObserver } from '../../global/helpers/resize-observer';
+
+enum NotificationIconName {
+  info = 'circle-information-small',
+  success = 'circle-tick-small',
+  warn = 'circle-exclamation-point-small',
+  error = 'circle-cross-small',
+}
+
+/**
+ * @slot title - Use this to provide a notification title (optional).
+ * @slot unnamed - Use this to provide the notification message.
+ */
+@Component({
+  shadow: true,
+  styleUrl: 'sbb-notification.scss',
+  tag: 'sbb-notification',
+})
+export class SbbNotification implements ComponentInterface {
+  /**
+   * The type of the notification.
+   */
+  @Prop({ reflect: true }) public type?: InterfaceNotificationAttributes['type'] = 'info';
+
+  /**
+   * Content of title.
+   */
+  @Prop() public titleContent?: string;
+
+  /**
+   * Level of title, will be rendered as heading tag (e.g. h3). Defaults to level 3.
+   */
+  @Prop() public titleLevel: InterfaceTitleAttributes['level'] = '3';
+
+  /**
+   * Whether the notification is readonly.
+   * In readonly mode, there is no dismiss button offered to the user.
+   */
+  @Prop({ reflect: true }) public readonly = false;
+
+  /**
+   * Whether the animation is enabled.
+   */
+  @Prop({ reflect: true, mutable: true }) public disableAnimation = false;
+
+  /**
+   * State of listed named slots, by indicating whether any element for a named slot is defined.
+   */
+  @State() private _namedSlots = createNamedSlotState('title');
+
+  /**
+   * The state of the notification.
+   */
+  @State() private _state: InterfaceNotificationAttributes['state'] = 'opened';
+
+  @State() private _iconName: NotificationIconName = NotificationIconName[this.type];
```

Comment:
I don‘t see the sense of having the icon as a state prop. Just use it in render and you can also remove watch on type